src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -59,22 +59,36 @@ bool IWorkspaceProjectContext.LastDesignTimeBuildSucceeded
                 _lastDesignTimeBuildSucceeded = value;
             }
         }
+
+        string IWorkspaceProjectContext.ObjOutputPath
+        {
+            get
+            {
+                return this.TryGetObjOutputPath();
+            }
+            set
+            {
+                SetOutputPathAndRelatedData(objOutputPath: value, binOutputPath: this.TryGetBinOutputPath());
+            }
+        }
+
         #endregion
 
         #region Options
-        public void SetCommandLineArguments(string commandLineForOptions)
+        public void SetOptions(string commandLineForOptions)
         {
             var commandLineArguments = SetArgumentsAndUpdateOptions(commandLineForOptions);
-            PostSetCommandLineArguments(commandLineArguments);
+            PostSetOptions(commandLineArguments);
         }
 
-        private void PostSetCommandLineArguments(CommandLineArguments commandLineArguments)
+        private void PostSetOptions(CommandLineArguments commandLineArguments)
         {
             // Invoke SetOutputPathAndRelatedData to update the project tracker bin path for this project, if required.
             if (commandLineArguments.OutputFileName != null && commandLineArguments.OutputDirectory != null)
             {
-                var newOutputPath = PathUtilities.CombinePathsUnchecked(commandLineArguments.OutputDirectory, commandLineArguments.OutputFileName);
-                SetOutputPathAndRelatedData(newOutputPath, hasSameBinAndObjOutputPaths: true);
+                var binOutputPath = PathUtilities.CombinePathsUnchecked(commandLineArguments.OutputDirectory, commandLineArguments.OutputFileName);
```

Comment:
I am wondering if I am doing this incorrectly. Should the output path from command line arguments correspond to the intermediate path instead of the binary output path? If so, I'll change the new property exposed on IWorkspaceProjectContext to be BinOutputPath instead of ObjOutputPath.